bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/StateFilterProfile.java

```diff
@@ -425,6 +433,12 @@ public boolean check(State input) {
 
                 rhs = Objects.requireNonNull(rhsState instanceof StringType ? rhsState.toString() : rhsState);
 
+                if (isDeltaCheck && rhs instanceof QuantityType rhsQty && lhs instanceof QuantityType lhsQty) {
```

Comment:
Hmm. Please look at the equivalent line in my PR #18144. There may be some overlap here.

Reply:
Yes I get that. This is more a not to myself than to you since I will need to work around it in my own PR. So let us proceed with your PR first, and I will handle the rest later.